server/src/main/java/org/elasticsearch/action/admin/cluster/snapshots/create/TransportCreateSnapshotAction.java

```diff
@@ -53,9 +53,7 @@ public TransportCreateSnapshotAction(TransportService transportService, ClusterS
 
     @Override
     protected String executor() {
-        // Using the generic instead of the snapshot threadpool here as the snapshot threadpool might be blocked on long running tasks
-        // which would block the request from getting an error response because of the ongoing task
-        return ThreadPool.Names.GENERIC;
+        return ThreadPool.Names.SAME;
```

Comment:
First step of creating a snapshot nowadays is putting the init state snapshot in the CS => forking to the CS thread right away anyway